### PR TITLE
Canvas layout: Add interactive SVG support

### PR DESCRIPTION
**Pull request by stefan-hoehn**
**This is major contribution to the fixed canvas layout** which allows an SVG background to become an active part of the page. Through this feature the SVG itself can become an active component of the page.

_Thanks to @florian-h05 for the intensive collaboration on this new feature._

It allows real embedding of an SVG if used as the background URL to become interactive:

image

- Embed SVG = ON: Loads the SVG into the HTML DOM, so the elements are available for the openHAB page.
- Embed SVG Flashing in Run-Mode in Run-Mode

image

Here is an **example** of a floorplan that has been implemented without any widget (note that widgets are still supported and are still recommended to be used) by showing some of the features:

image

https://github.com/user-attachments/assets/aa8f1094-5d53-43bc-a31e-ebcd872b6181

- The violet "led strip" can directly be clicked to turn on and off the light
- The SVG element that represents that strip changes its color based on the real color of the light strip

- The blue bar is a window whose rollershutter is partially closed. The blue color changes its transparency based on the opening percentage of the rollershutter (it becomes completely gray like the wall when the rollershutter is closed)
image

- All other elements in the picture are completely contained in the SVG (some of which could have also been done via a widget but they serve as an example of the possible features.

- State information can be directly written into an SVG-Element like the temperature here
 
image

- Different representations can be achieved by enabling / disabling parts of the element like the bulb here:

OFF / ON (glow around the bulb, which could be even animated within the SVG)

image image

- Style changes can be trigged based on an ON/STATE. Even animations that are contained in the SVG can be triggered via dynamic style changes manipulating the SVG like the following slider button that has been implemented within the SVG or the blinking astronaut on the right in the video. This allows endless ideas (e.g. energy flow within a house)

- SVG that should become active elements for openHAB are marked with an "openhab" attribute (in the SVG editor). These will the be detected in the openHAB editor and made available for configuration:

https://github.com/user-attachments/assets/f371afe5-157c-4712-af15-5d97a16248f5

and by clicking on it the element can be configured similarly like a widget:

https://github.com/user-attachments/assets/84de771d-a621-4cad-91c3-23108e5d7267

Keep in mind: it is not a replacement for widgets ... widgets can and should still be used but it allows the background itself to be as interactive as a widget.

PS: A comprehensive documentation will come soon

### CODE REVIEW

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+          if (!element) {
+            console.warn(`Element ${svgElement} is a group element but has no containing element with the attribute "flash"`)
+            return
+          }
+          if (state === 'ON' || stateType === 'HSB') {
+            if (useProxy && svgElementConfig.stateAsOpacity) { // we use the flash element
+              let opacity = (state === 'ON') ? 1 : 0
+              opacity = (svgElementConfig.invertStateOpacity) ? 1 - opacity : opacity
+              opacity = (opacity < svgElementConfig.stateMinOpacity) ? svgElementConfig.stateMinOpacity : opacity
+              // TODO: use fill-opacity if fill not available
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:308]: What about this to do?

**Review by florian-h05 — COMMENTED** [review]
I have rebased and refactored it a bit to use a better approach for widget actions than my first solution. It now also used Framework7 color conversion methods instead of your HSB to RGB implementation.

I have changed a few minor things in your code, code now LGTM!

Can you please test it @stefan-hoehn?

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js (hunk 1 above)
Comment (stefan-hoehn) [reply bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:308]: This is meant for later improvement of SVG usage but I need to look further into this topic in which cases that might happen at all.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:308]: Fine with me :+1:

**Comment by stefan-hoehn** [comment]
@florian-h05 First tests look pretty promising. I had to setup on item again but other than that all configured actions and items work well. I will do a bit of more testing today but I am pretty sure this finally can be merged.

**Comment by stefan-hoehn** [comment]
LGTM, haven't found any issue, good job, Florian.

**Review by florian-h05 — APPROVED** [review]

**Pull request merged**